src/app/planner/planner/planner.component.ts

```diff
@@ -651,4 +653,41 @@ export class PlannerComponent extends SubscriptionsComponent implements OnInit {
 	sidebarClose() {
 		this.sidebarCollapsed = true;
 	}
+
+	// modal date range selection
+
+	hoveredDate: NgbDate | null = null;
+
+	fromDate: NgbDate | null;
+	toDate: NgbDate | null;
+
+	onDateSelection(date: NgbDate) {
+		if (!this.fromDate && !this.toDate) {
+		  this.fromDate = date;
+		} else if (this.fromDate && !this.toDate && date && date.after(this.fromDate)) {
+		  this.toDate = date;
+		  this.createEventModel.dates = [this.dateAdapter.toModel(this.fromDate)!!, this.dateAdapter.toModel(this.toDate)!!]
```

Comment:
What's with the double bangs? Non-null assertion is only one

Reply:
hmmm, let me try